adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -65,61 +73,167 @@ private GenericContainer<?> createAndStartCollector(String configFilePath, Strin
+            fileWriter.write("Second Message after the collector is stopped" + "\n");
+            fileWriter.write("Third Message after the collector is stopped" + "\n");
+            fileWriter.flush();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        fileWriter.close();
+       //restarting the collector
+        collector.start();
+
+        validateLogs(logStreamName , logFilePaths, false);
+        collector.stop();
+    }
+
+    @Test
+    void testFileRotation() throws Exception {
+        String logStreamName = "fileRotation-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-fileRotation.yaml", logStreamName);
+
+        Thread.sleep(5000);
+
+        // Create and write data to File A
+        File tempFile = new File(logDirectory.toString(), "testlogA.log");
+        try (FileWriter fileWriter = new FileWriter(tempFile)) {
+            fileWriter.write("Message in File A" + "\n");
+            fileWriter.flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        Thread.sleep(5000);
+
+        File renameFile = new File(logDirectory.toString(), "testlogA-1234.log");
+        tempFile.renameTo(renameFile);
+
+        File tempFileB = new File(logDirectory.toString(), "testlogA.log");
+        try (FileWriter newfileWriter = new FileWriter(tempFileB)) {
+            newfileWriter.write("Message in renamed file - line 1" + "\n");
+            newfileWriter.write("Message in renamed file - line 2" + "\n");
+            newfileWriter.write("Message in renamed file - line 3" + "\n");
+            newfileWriter.flush();
+            newfileWriter.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File B.", e);
+        }
+        Thread.sleep(5000);
+
+        List<String> logFilePaths = new ArrayList<>();
+        String expectedLogPath = logDirectory.toString();
+        logFilePaths.add(expectedLogPath + "/testlogA-1234.log");
+        logFilePaths.add(expectedLogPath + "/testlogA.log");
+
+        validateLogs(logStreamName, logFilePaths, false);
+
+        collector.stop();
+    }
+
+
+    void validateLogs(String testLogStreamName, List<String> logFilePaths, boolean areResourceFiles) throws Exception {
         var lines = new HashSet<String>();
 
-        try (InputStream inputStream = getClass().getResourceAsStream(logFilePath);
-             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
-            String line;
-            while ((line = reader.readLine()) != null) {
-                lines.add(line);
+        for (String logFilePath : logFilePaths) {
+            InputStream inputStream;
+            //Check whether the filePath is from resource folder.
+            if (areResourceFiles) {
```

Comment:
https://luzkan.github.io/smells/flag-argument

Reply:
why don't you always pass as parameter to this method a list of InputStreams?

Reply:
Agreed, added a check within the method, to let it decide